### DISCUSSION TITLE
Issue with debug model for ner_demo_update project

### DISCUSSION THREAD

**Discussion by lzmodel** [explosion/spaCy#12254] [Help: Coding & Implementations] [answered]
## How to reproduce the behaviour

1. Executed `spacy project run create-config` for [ner_demo_update](https://github.com/explosion/projects/tree/v3/pipelines/ner_demo_update) project
2. Run `spacy debug model configs/config.cfg ner -P0 -P1 --gpu-id 0`
========================== STEP 0 - before training ==========================
...
ℹ Layer 66: model ID 175: 'maxout'
ℹ Layer 67: model ID 176: 'layernorm'

✘ Could not initialize the model: you'll have to provide a valid
'train_corpus' argument in the config file.

3. Executed  `spacy project run train` - successfully
4. Edited config.cfg file by adding params like in command line `--paths.train corpus/train.spacy --paths.dev corpus/dev.spacy --training.eval_frequency 10 --training.max_steps 500 --training.patience 50` and `run spacy debug model` - error remains the same.

I tried to edit this config.cfg with my data and by deleting some components from pipeline, but i can not know exactly if these error  mine or from original config.
## Your Environment
- **spaCy version:** 3.4.1
- **Platform:** Linux-5.15.0-48-generic-x86_64-with-glibc2.35
- **Python version:** 3.9.13
- **Pipelines:** en_core_web_sm (3.4.1), en_core_web_lg (3.4.1), en_core_web_trf (3.4.1)
- ** Environment Information:** conda env

**Comment by danieldk** [comment]
I can't reproduce this issue. From a fresh checkout of the project:

```
$ spacy project run create-config
$ spacy debug model configs/config.cfg ner -P0 -P1 --gpu-id 0
[...]
ℹ Layer 66: model ID 175: 'maxout'
ℹ Layer 67: model ID 176: 'layernorm'
✔ Succesfully ended analysis - model looks good.
```

What changes did you make to the configuration? What do you have in the `paths` section? Could you also check if `debug data` works for the original project configuration?

**Comment by lzmodel** [comment]
I just did the following:
1) deleted the [ner_demo_update](https://github.com/explosion/projects/tree/v3/pipelines/ner_demo_update) project completely
2) reinstalled the project
3) spacy project run create-config
4) run `python -m spacy debug model configs/config.cfg ner -P0 -P1 --gpu-id 0`
Result is the same:
```
ℹ Layer 65: model ID 168: 'layernorm'
ℹ Layer 66: model ID 175: 'maxout'
ℹ Layer 67: model ID 176: 'layernorm'

✘ Could not initialize the model: you'll have to provide a valid
'train_corpus' argument in the config file.
```

**Comment by adrianeboyd** [comment, marked as answer]
I think you're just missing the overrides for the corpus paths in your command (something like `--paths.train train.spacy --paths.dev dev.spacy`).

**Comment by lzmodel** [comment]
@adrianeboyd How to check paths correctly? 
1) I edited config.cfg by adding
```
train = "./corpus/train.spacy"
dev = "./corpus/dev.spacy"
```
2) Run `spacy debug data ./configs/config.cfg` - OK.
3) Run `spacy debug model configs/config.cfg ner -P0 -P1 --gpu-id 0` - OK
4) I changed the directory to my data, that i used for training model from scratch many times (data exist exactly) - now i have another error:
```
...
File "spacy/pipeline/tagger.pyx", line 297, in spacy.pipeline.tagger.Tagger.initialize
  File "spacy/pipeline/tagger.pyx", line 322, in spacy.pipeline.tagger.Tagger.add_label
  File "spacy/pipeline/trainable_pipe.pyx", line 224, in spacy.pipeline.trainable_pipe.TrainablePipe._allow_extra_label
ValueError: [E922] Component 'tagger' has been initialized with an output dimension of 50 - cannot add any more labels.
```
The Path to the data is correct. To check it I tried to corrupt it and got another message "FileNotFoundError: [Errno 2] No such file or directory".

**Comment by rmitsch** [comment]
Moving this to discussions, as this seems to be a usage-related question rather than an issue with spaCy per se.

**Comment by rmitsch** [comment]
@lzmodel Please provide your complete config, it'll help us reproducing your issue.

**Reply by lzmodel** [reply]
Hello, @rmitsch 
```[paths]
train = "./train/data/v23/train.spacy"
dev = "./train/data/v23/dev.spacy"
vectors = null
init_tok2vec = null

[system]
gpu_allocator = null
seed = 0

[nlp]
lang = "en"
pipeline = ["tok2vec","tagger","parser","senter","attribute_ruler","lemmatizer","ner"]
disabled = ["senter"]
before_creation = null
after_creation = null
after_pipeline_creation = null
batch_size = 256
tokenizer = {"@tokenizers":"spacy.Tokenizer.v1"}

[components]

[components.attribute_ruler]
source = "en_core_web_lg"

[components.lemmatizer]
source = "en_core_web_lg"

[components.ner]
source = "en_core_web_lg"
replace_listeners = ["model.tok2vec"]

[components.parser]
source = "en_core_web_lg"

[components.senter]
source = "en_core_web_lg"

[components.tagger]
source = "en_core_web_lg"

[components.tok2vec]
source = "en_core_web_lg"

[corpora]

[corpora.dev]
@readers = "spacy.Corpus.v1"
path = ${paths.dev}
gold_preproc = false
max_length = 0
limit = 0
augmenter = null

[corpora.train]
@readers = "spacy.Corpus.v1"
path = ${paths.train}
gold_preproc = false
max_length = 0
limit = 0
augmenter = null

[training]
train_corpus = "corpora.train"
dev_corpus = "corpora.dev"
seed = ${system:seed}
gpu_allocator = ${system:gpu_allocator}
dropout = 0.1
accumulate_gradient = 1
patience = 5000
max_epochs = 0
max_steps = 100000
eval_frequency = 1000
frozen_components = ["tok2vec","tagger","parser","senter","attribute_ruler","lemmatizer"]
before_to_disk = null
annotating_components = []

[training.batcher]
@batchers = "spacy.batch_by_words.v1"
discard_oversize = false
tolerance = 0.2
get_length = null

[training.batcher.size]
@schedules = "compounding.v1"
start = 100
stop = 1000
compound = 1.001
t = 0.0

[training.logger]
@loggers = "spacy.ConsoleLogger.v1"
progress_bar = false

[training.optimizer]
@optimizers = "Adam.v1"
beta1 = 0.9
beta2 = 0.999
L2_is_weight_decay = true
L2 = 0.01
grad_clip = 1.0
use_averages = true
eps = 0.00000001
learn_rate = 0.001

[training.score_weights]
tag_acc = 0.16
dep_uas = 0.0
dep_las = 0.16
dep_las_per_type = null
sents_p = null
sents_r = null
sents_f = 0.02
lemma_acc = 0.5
ents_f = 0.16
ents_p = 0.0
ents_r = 0.0
ents_per_type = null
speed = 0.0

[pretraining]

[initialize]
vocab_data = null
vectors = null
init_tok2vec = ${paths.init_tok2vec}
after_init = null
lookups = null

[initialize.before_init]
@callbacks = "spacy.copy_from_base_model.v1"
tokenizer = "en_core_web_lg"
vocab = "en_core_web_lg"

[initialize.components]

[initialize.components.ner]

[initialize.components.ner.labels]
@readers = "spacy.read_labels.v1"
path = "corpus/labels/ner.json"
require = false

[initialize.components.parser]

[initialize.components.parser.labels]
@readers = "spacy.read_labels.v1"
path = "corpus/labels/parser.json"
require = false

[initialize.components.tagger]

[initialize.components.tagger.labels]
@readers = "spacy.read_labels.v1"
path = "corpus/labels/tagger.json"
require = false

[initialize.tokenizer]
```

**Reply by rmitsch** [reply]
> I changed the directory to my data, that i used for training model from scratch many times (data exist exactly)

This is the data you're talking about, right? How did you generate it?
```
train = "./train/data/v23/train.spacy"
dev = "./train/data/v23/dev.spacy"
```

**Reply by lzmodel** [reply]
It's my own data, labeled with Label Studio, exported and converted into spacy format. With another config I can debug it:
```
spacy debug data ./train/cfg_spc/cpu10v23-lg-tok.cfg

============================ Data file validation ============================
✔ Pipeline can be initialized with data
✔ Corpus is loadable

=============================== Training stats ===============================
Language: en
Training pipeline: tok2vec, ner
758 training docs
187 evaluation docs
⚠ 1 training examples also in evaluation data
⚠ Low number of examples to train a new pipeline (758)

============================== Vocab & Vectors ==============================
ℹ 395493 total word(s) in the data (11877 unique)
ℹ 514157 vectors (514157 unique keys, 300 dimensions)
⚠ 4665 words in training data without vectors (1%)

========================== Named Entity Recognition ==========================
ℹ 10 label(s)
0 missing value(s) (tokens with '-' label)
✔ Good amount of examples for all labels
✔ Examples without occurrences available for all labels
✔ No entities consisting of or starting/ending with whitespace
✔ No entities crossing sentence boundaries

================================== Summary ==================================
✔ 6 checks passed
⚠ 3 warnings
```

**Reply by rmitsch** [reply]
> With another config I can debug it: ...

You're saying "another config". Previously you mentioned that debugging worked for you:

> 3. Run spacy debug model configs/config.cfg ner -P0 -P1 --gpu-id 0 - OK ...

So is the `debug` command working for you with the config you posted above? Also, can you share the data you're using (at least a minimal version thereof)?

**Reply by lzmodel** [reply]
Sorry, @rmitsch Raphael, I missed and did not write what I changed yet, because I considered it not significant - the size of the model. In the file `project.yml`, I read that 

> for en_core_web_trf or other trf pipeline, use spacy-transformers>=1.1.1.

So, if you specify pipeline: "en_core_web_lg" or pipeline: "en_core_web_trf" in `project.yml`, run `create config`, than `spacy debug model` gives an error
```
✘ Could not initialize the model: you'll have to provide a valid
'train_corpus' argument in the config file
```
I checked it with data from the project ner_demo_update

```
train = "./corpus/train.spacy"
dev = "./corpus/dev.spacy"
```
Answers to your questions:

> 1. So is the debug command working for you with the config you posted above?

Config posted above contains updating of en_core_web_lg model. It does not work. In paragraph 3, I wrote about a completely different model config for training from scratch.

> 2. Also, can you share the data you're using (at least a minimal version thereof)?

Yes, I can. But as you can see above I found the shortest way to reproduce error in debugging this model

**Reply by rmitsch** [reply]
Let me rehash this, because I'm not sure I follow. So if you
1. clone the `ner_demo_update` project,
2. use the config you posted [here](https://github.com/explosion/spaCy/discussions/12254#discussioncomment-4923286), _but with the original data_ (i. e. the train and dev paths are pointing to the original files, not your self-labeled ones),
3. specify `en_core_web_lg` or `en_core_web_trf` (but not e. g. `en_core_web_sm`) in your `project.yml`,
4. and finally run `spacy debug model`,
then you run into
```shell
python -m spacy debug model configs/config.cfg ner -P0 -P1 --gpu-id 0
```
Is this correct?

**Reply by lzmodel** [reply]
Sorry Raphael @rmitsch, it's my fought with the corpus paths. Thank you for your time.
